src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala

```diff
@@ -543,13 +543,13 @@ object DatabaseConversions extends LazyLogging {
     * We get back an empty List, where not applicable
     * We define a typeclass/trait that encodes the availability of balance updates, to reuse it for operations as well as blocks
     * @tparam T the source type that contains the balance updates values
-    * @tparam Label the source Label type, which has a `Show` contraint, to be representable as a string
+    * @tparam Label the source Label type, which has a `Show` constraint, to be representable as a string
     * @param balances an optic instance that lets extract the Map of tagged updates from the source
     * @param hashing an optic instance to extract an optional reference hash value
     */
   implicit def anyToBalanceUpdates[T, Label](
       implicit
-      balances: Getter[T, Map[Label, List[OperationMetadata.BalanceUpdate]]],
+      balances: Getter[T, Map[Label, List[BlockTagged[OperationMetadata.BalanceUpdate]]]],
```

Comment:
Wouldn't it be more efficient to "tag" the `Label` here?
That would be made once per block, for operations and header, instead of replicating the same data for each update.

```scala
balances: Getter[T, Map[BlockTagged[Label], List[OperationMetadata.BalanceUpdate]]],
```
and then we pattern match directly on the key.

It should simplify also the definition of instances [here](https://github.com/Cryptonomic/Conseil/pull/779/files#diff-9a63ae02048772847f4ed11b44a80206R37)
WDYT?